java/arcs/core/data/expression/Expression.kt

```diff
@@ -390,6 +393,16 @@ sealed class Expression<out T> {
         override fun <Result> accept(visitor: Visitor<Result>): Result = visitor.visit(this)
         override fun toString() = this.stringify()
     }
+
+    /**
+     * Represents an expression that must be parsed and then evaluated.
+     */
+    data class EvalExpression<T>(
```

Comment:
What is the benefit of constructing an EvalExpression node vs: Just parsing the expression when we want to convert string -> Expression ? Maybe other angle would be: When do you think we should use it?